apps/dashboard/src/components/conditions-editor/conditions-editor-context.tsx

```diff
@@ -0,0 +1,65 @@
+import { createContext, useCallback, useContext, useMemo, useState } from 'react';
+import { RuleGroupType, RuleType, remove, add, isRuleGroup, RuleGroupTypeAny, Path } from 'react-querybuilder';
+
+import { ConditionsEditorContextType } from './types';
+
+export const ConditionsEditorContext = createContext<ConditionsEditorContextType>({
+  query: { combinator: 'and', rules: [] },
+  setQuery: () => {},
+  removeRuleOrGroup: () => {},
+  cloneRuleOrGroup: () => {},
+  getParentGroup: () => null,
+});
+
+export function ConditionsEditorProvider({ children }: { children: React.ReactNode }) {
```

Comment:
The context provider for the editor. It stores the query, allows to remove and clone condition/group, and get the nearest parent.